backend/api/views/quiz.py

```diff
@@ -1,10 +1,80 @@
-from rest_framework.decorators import api_view
+from django.http import Http404
+from rest_framework.views import APIView
+from rest_framework.request import Request
+from rest_framework import status
+from pymongo.collection import ReturnDocument
 
+from ..validators import SchemaValidator
 from ..client import client
-from ..utils import JsonResponse
+from ..utils import JsonResponse, get_sort_order_from_request, get_new_id
+from ..exceptions import InvalidData
 
 
-@api_view(["GET"])
-def quiz_list(request):
-    if request.method == "GET":
-        return JsonResponse(data=client.collection("quizzes").find())
+class QuizList(APIView):
+    __quizzes = client.collection("quizzes")
+    __validator = SchemaValidator("quizzes")
```

Comment:
Do you know how I can avoid duplicating these two lines in both classes ?

Reply:
create a base class that herits from APIView

Reply:
I've already tried it, but I was getting the error `'QuizList' object has no attribute '_QuizList__quizzes'` when fetching `/api/quizzes/` 😭